src/keysym.c

```diff
@@ -270,16 +270,18 @@ xkb_keysym_from_name(const char *name, enum xkb_keysym_flags flags)
     * Find the correct keysym for case-insensitive match.
     *
     * The name_to_keysym table is sorted by istrcmp(). So the binary
-    * search may return _any_ of all possible case-insensitive duplicates. This
-    * code searches the entry, all previous and all next entries that match by
-    * case-insensitive comparison and returns the "best" case-insensitive
-    * match.
+    * search may return _any_ of all possible case-insensitive duplicates. The
+    * duplicates are sorted so that the "best" case-insensitive  match comes
+    * last. So the code searches the entry and all next entries that match by
+    * case-insensitive comparison and returns the "best" case-insensitive match.
     *
-    * The "best" case-insensitive match is the lower-case keysym which we find
-    * with the help of xkb_keysym_is_lower(). The only keysyms that only differ
-    * by letter-case are keysyms that are available as lower-case and
-    * upper-case variant (like KEY_a and KEY_A). So returning the first
-    * lower-case match is enough in this case.
+    * The "best" case-insensitive match is the lower-case keysym name. The only
+    * keysyms names that only differ by letter-case are keysyms that are
+    * available as “small” and “big” variants. For example:
```

Comment:
I have found one more example of ambiguous keycode names: `XF86ScreenSaver` and `XF86Screensaver`, not sure which one is the right one to return here (maybe it doesn't matter?)

Reply:
Check [this test file](https://github.com/xkbcommon/libxkbcommon/blob/7194d5a19eb79ba2b29081b7fcd38737f1c12fd5/test/keysym.h) to find out. Doc can be improved, comment welcome!

Reply:
I'd say "most keysym names that only differ...", but that's a minor nitpick.